crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -423,6 +428,11 @@ where
         to_block: u64,
     ) -> Result<Vec<Log>, FilterError> {
         trace!(target: "rpc::eth::filter", from=from_block, to=to_block, ?filter, "finding logs in range");
+        let is_multi_block_range = from_block != to_block;
+
+        if is_multi_block_range && to_block - from_block > self.max_blocks_per_filter {
```

Comment:
nit: do we need this `is_multi_block_range` check? If `to_block == from_block`, then `to_block - from_block == 0` and this condition always evaluates to `false`.

```suggestion
        if to_block - from_block > self.max_blocks_per_filter {
```

Reply:
right, we don't